packages/outputarea/src/widget.ts

```diff
@@ -429,52 +450,49 @@ export class OutputArea extends Widget {
    * @param model - The model of the output to be inserted.
    */
   private _insertOutput(index: number, model: IOutputModel): void {
-    if (index === 0) {
-      this.trimmedOutputModels = new Array<IOutputModel>();
+    if (index > this._maxNumberOutputs) {
+      return;
     }
-    if (index === this.maxNumberOutputs && this.maxNumberOutputs !== 0) {
+
+    if (index === this._maxNumberOutputs) {
       // TODO Improve style of the display message.
-      const separatorModel = this.model.contentFactory.createOutputModel({
+      model = this.model.contentFactory.createOutputModel({
         value: {
           output_type: 'display_data',
           data: {
             'text/html': `
               <a style="margin: 10px; text-decoration: none; cursor: pointer;">
                 <pre>Output of this cell has been trimmed on the initial display.</pre>
-                <pre>Displaying the first ${this.maxNumberOutputs} top outputs.</pre>
+                <pre>Displaying the first ${this._maxNumberOutputs} top outputs.</pre>
                 <pre>Click on this message to get the complete output.</pre>
               </a>
               `
           }
         }
       });
-      const onClick = () => this._showTrimmedOutputs();
-      const separator = this.createOutputItem(separatorModel);
-      separator!.node.addEventListener('click', onClick);
-      const layout = this.layout as PanelLayout;
-      layout.insertWidget(this.headEndIndex, separator!);
-    }
-    const output = this._createOutput(model);
-    const layout = this.layout as PanelLayout;
-    if (index < this.maxNumberOutputs || this.maxNumberOutputs === 0) {
-      layout.insertWidget(index, output);
-    }
-    if (index >= this.maxNumberOutputs && this.maxNumberOutputs !== 0) {
-      this.trimmedOutputModels.push(model);
     }
-    if (!this._outputTracker.has(output)) {
-      void this._outputTracker.add(output);
-    }
-  }
 
-  private _createOutput(model: IOutputModel): Widget {
     let output = this.createOutputItem(model);
     if (output) {
       output.toggleClass(EXECUTE_CLASS, model.executionCount !== null);
     } else {
       output = new Widget();
     }
-    return output;
+
+    if (!this._outputTracker.has(output)) {
+      void this._outputTracker.add(output);
+    }
+
+    const layout = this.layout as PanelLayout;
+    layout.insertWidget(index, output);
+
+    if (index === this._maxNumberOutputs) {
+      output.node.addEventListener('click', () => {
```

Comment:
Can this event listener be added to the `OutputArea` class in the normal Lumino way instead of on-the-fly here?